src/texture_wrapper.rs

```diff
@@ -99,8 +183,73 @@ impl TexWrap {
         gfx: &mut Graphics,
         settings: &PersistentSettings,
         image: &DynamicImage,
-    ) -> Option<TexWrap> {
-        Self::gen_from_dynamic_image(gfx, settings, image, Self::gen_texture_standard)
+        swizzle_mask: Mat4,
+        offset_vec: Vec4,
+    ) -> Result<TexWrap, String> {
+        Self::gen_from_dynamic_image(
+            gfx,
+            settings,
+            image,
+            Self::gen_texture_standard,
+            swizzle_mask,
+            offset_vec,
+        )
+    }
+
+    fn gen_uniform_buffer_swizzle_mask(
+        gfx: &mut Graphics,
+        swizzle_mask: Mat4,
+        offset_vec: Vec4,
+    ) -> Result<(Buffer, Buffer), String> {
+        let uniform_swizzle_mask = gfx
+            .create_uniform_buffer(1, "SwizzleMask")
+            .with_data(&swizzle_mask)
+            .build();
+
+        match uniform_swizzle_mask {
```

Comment:
Instead of the match block, you should be able to just use `?` after line 207, for exampe
```
let uniform_swizzle_mask = gfx
    .create_uniform_buffer(1, "SwizzleMask")
    .with_data(&swizzle_mask)
    .build()?;
```

If the error types don't match, you can use `.map_err()` to fix that.